src/sugar.jl

```diff
@@ -188,12 +189,16 @@ julia> set(t, (@lens _[1]), "1")
 
 """
 macro lens(ex)
+    atlens_impl(ex)
```

Comment:
I guess we don't need to expose `atlens_impl` as people can just do

```julia
macro mylens(ex)
    :(mylenstransform(@lens $ex))
end
```

? Or maybe with some more `$` and `esc`. :)

Reply:
It is nice, that this is a one liner. On the other hand the fact that you are asking about `$` and `esc` shows that it is somewhat non trivial to get isolation right. So I left `lensmacro` alive.